### PR TITLE
Add size limit for uids cookie

### PR DESCRIPTION
**Pull request by RossGoncharuk**
Added a size limit for UIDs Cookie through new configuration property - `host-cookie.max-cookie-size-bytes` with the default value is `0`, which means the size limit is disabled.

### CODE REVIEW

File: src/main/java/org/prebid/server/spring/config/ServiceConfiguration.java [hunk 1]
```diff
@@ -234,10 +234,17 @@ UidsCookieService uidsCookieService(
             @Value("${host-cookie.family:#{null}}") String hostCookieFamily,
             @Value("${host-cookie.cookie-name:#{null}}") String hostCookieName,
             @Value("${host-cookie.domain:#{null}}") String hostCookieDomain,
-            @Value("${host-cookie.ttl-days}") Integer ttlDays) {
+            @Value("${host-cookie.ttl-days}") Integer ttlDays,
+            @Value("${host-cookie.max-cookie-size-bytes}") Integer maxCookieSizeBytes) {
+
+        final int minCookieSizeBytes = 500;
```
Comment (schernysh) [inline src/main/java/org/prebid/server/spring/config/ServiceConfiguration.java]: Would like to suggest to move this validation to `UidsCookieService`'s constructor.

File: src/main/java/org/prebid/server/spring/config/ServiceConfiguration.java [hunk 2]
```diff
@@ -234,10 +234,17 @@ UidsCookieService uidsCookieService(
...
+        if (maxCookieSizeBytes != 0 && maxCookieSizeBytes < minCookieSizeBytes) {
+            throw new IllegalArgumentException(String.format(
+                    "Configured cookie size is less than allowed minimum size of %s", maxCookieSizeBytes));
```
Comment (schernysh) [inline src/main/java/org/prebid/server/spring/config/ServiceConfiguration.java]: `%s` have to be replaced with `%d` as substitution variable is numerical.

File: src/main/java/org/prebid/server/cookie/UidsCookieService.java [hunk 3]
```diff
@@ -124,6 +139,15 @@ public Cookie toCookie(UidsCookie uidsCookie) {
         return cookie;
     }
 
+    /**
+     * Returns the Uid with the closest expiration date, e.i. the one that will expire sooner.
+     */
+    private static Map.Entry<String, UidWithExpiry> getClosestExpiration(Map.Entry<String, UidWithExpiry> first,
+                                                                         Map.Entry<String, UidWithExpiry> second) {
+        return first.getValue().getExpires().toEpochSecond() < second.getValue().getExpires().toEpochSecond()
```
Comment (schernysh) [inline src/main/java/org/prebid/server/cookie/UidsCookieService.java]: Instead of using `toEpochSecond` consider using `isBefore`

Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/cookie/UidsCookieService.java]: Good call, thanks!

File: src/main/java/org/prebid/server/cookie/UidsCookieService.java [hunk 4]
```diff
@@ -30,22 +30,31 @@
...
     private final String hostCookieFamily;
     private final String hostCookieName;
     private final String hostCookieDomain;
     private final Long ttlSeconds;
+    private final Integer maxCookieSizeBytes;
 
     public UidsCookieService(String optOutCookieName, String optOutCookieValue, String hostCookieFamily,
-                             String hostCookieName, String hostCookieDomain, Integer ttlDays) {
+                             String hostCookieName, String hostCookieDomain, Integer ttlDays,
+                             Integer maxCookieSizeBytes) {
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/cookie/UidsCookieService.java]: Can be primitive.

**Pull request merged**